src/containers/App/selectors.js

```diff
@@ -101,24 +101,23 @@ const makeSelectUserCanAccess = createSelector(
     }
 );
 
-const makeSelectLoading = () => createSelector(selectGlobal, globalState => globalState.get('loading'));
+export const makeSelectLoading = () => createSelector(selectGlobal, globalState => globalState.get('loading'));
 
-const makeSelectError = () => createSelector(selectGlobal, globalState => globalState.get('error'));
+export const makeSelectError = () => createSelector(selectGlobal, globalState => globalState.get('error'));
 
-const makeSelectNotification = () =>
+export const makeSelectNotification = () =>
   createSelector(selectGlobal, globalState => globalState.get('notification').toJS());
 
-const makeSelectSearchQuery = createSelector(selectGlobal, globalState => globalState.get('searchQuery'));
-
-export {
-  makeSelectError,
-  makeSelectLoading,
-  makeSelectNotification,
-  makeSelectUser,
-  makeSelectUserCan,
-  makeSelectUserCanAccess,
-  makeSelectUserPermissionCodeNames,
-  makeSelectUserPermissions,
-  selectGlobal,
-  makeSelectSearchQuery,
-};
+export const makeSelectSearchQuery = createSelector(selectGlobal, globalState => globalState.get('searchQuery'));
+
+export const makeSelectSources = createSelector(selectGlobal, globalState =>
+  globalState.get('sources').size
+    ? globalState
+      .get('sources')
+      .toJS()
+      .map(source => ({
+        key: String(source.id),
+        value: source.name,
+      }))
+    : null
```

Comment:
Are components that rely on the return value of this selector able to deal with the value `null` instead of getting an array of values?

Reply:
Yes, and just added/updated some unit tests to reflect that.